wordpress-mobile/WordPress-Android
# [Jetpack Focus] Implement Local User Flags ContentProvider and Resolver

Fixes #17144 #17205 #17144

To test:
1 - Install WordPress and log in;
2 - Set the flag `isFeatureFlagEnabled` to `true` on `SharedLoginResolver` **and** `UserFlagsResolver.kt` (this cannot be done using the `Debug settings` screen because that is only available when the user is already logged in);
2 - Install Jetpack using the same variant (e.g. if installed WordPress version was `jalapenoDebug`, install Jetpack `jalapenoDebug`);
3 - Jetpack should login automatically after launching, and home screen should be shown;
4 - Check SharedPreferences in Jetpack.

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and unit tests

3. What automated tests I added (or what prevented me from doing so)
`SharedLoginResolverTest.kt`, `JetpackLocalUserFlagsFlagTest.kt`, `UserFlagsAnalyticsTrackerTest.kt` and `UserFlagsResolverTest.kt`

PR submission checklist:
